include/seqan3/alignment/pairwise/alignment_configurator.hpp

```diff
@@ -282,10 +286,61 @@ struct alignment_configurator
         if constexpr (config_t::template exists<align_cfg::band>())
             throw invalid_alignment_configuration{"Banded alignments are yet not supported."};
 
-        if constexpr (config_t::template exists<align_cfg::aligned_ends>())
-            throw invalid_alignment_configuration{"Free end gaps is not yet supported."};
+        // ----------------------------------------------------------------------------
+        // Configure semi-global alignment
+        // ----------------------------------------------------------------------------
+
+        // Get the value for the sequence ends configuration.
+        auto align_ends_cfg = cfg.template value_or<align_cfg::aligned_ends>(align_cfg::none_ends_free);
+        using align_ends_cfg_t = remove_cvref_t<decltype(align_ends_cfg)>;
+
+        auto configure_with_semi_global = [&](auto is_semi_global)
+        {
+
+            struct edit_traits_type
+            {
+                using word_type           [[maybe_unused]] = uint_fast64_t;
```

Comment:
Do I need maybe unused in struct member variables? :o

Reply:
Well inside of the local trait the compiler (only gcc-7 btw) would emit an unused variable warning